qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -287,3 +703,281 @@ def use_high_resolution_adc(self) -> None:
+                else:
+                    raise AssertionError(
+                        "Polarity of start and end is not "
+                        "same.")
+
+        def linear_sweep(start: float, end: float, steps: int) -> tuple:
+            sweep_val = np.linspace(start, end, steps)
+            return tuple(sweep_val)
+
+        def log_sweep(start: float, end: float, steps: int) -> tuple:
+            sweep_val = np.logspace(np.log10(start), np.log10(end), steps)
+            return tuple(sweep_val)
+
+        def linear_2way_sweep(start: float, end: float,
+                              steps: int) -> tuple:
+            if steps % 2 == 0:
+                half_list = list(np.linspace(start, end, steps // 2))
+                sweep_val = half_list + half_list[::-1]
+            else:
+                half_list = list(np.linspace(start, end, steps // 2,
+                                             endpoint=False))
+                sweep_val = half_list + [end] + half_list[::-1]
+            return tuple(sweep_val)
+
+        def log_2way_sweep(start: float, end: float, steps: int) -> tuple:
+            if steps % 2 == 0:
+                half_list = list(
+                    np.logspace(np.log10(start), np.log10(end),
+                                steps // 2))
+                sweep_val = half_list + half_list[::-1]
+            else:
+                half_list = list(
+                    np.logspace(np.log10(start), np.log10(end),
+                                steps // 2, endpoint=False))
+                sweep_val = half_list + [end] + half_list[::-1]
+            return tuple(sweep_val)
+
+        modes = {1: linear_sweep,
+                 2: log_sweep,
+                 3: linear_2way_sweep,
+                 4: log_2way_sweep}
+
+        return modes[sweep_mode](start_value, end_value, step_value)
+
+
+class IVSweepMeasurement(MultiParameter):
+    """
+    IV sweep measurement outputs a list of primary and secondary
+    parameter.
+
+    Args:
+        name: Name of the Parameter.
+        instrument: Instrument to which this parameter communicates to.
+    """
+
+    def __init__(self, name, instrument, **kwargs):
+        super().__init__(
+            name,
+            names=tuple(['gate_current', 'source_drain_current']),
+            units=tuple(['A', 'A']),
+            labels=tuple(['Gate Current', 'Source Drain Current']),
+            shapes=((1,),) * 2,
+            setpoint_names=(('Voltage',),) * 2,
+            setpoint_labels=(('Voltage',),) * 2,
+            setpoint_units=(('V',),) * 2,
+            **kwargs)
+        self._instrument = instrument
+        self.data = _FMTResponse(None, None, None, None)
+        self.param1 = _FMTResponse(None, None, None, None)
+        self.param2 = _FMTResponse(None, None, None, None)
+        self.ac_voltage = _FMTResponse(None, None, None, None)
+        self.dc_voltage = _FMTResponse(None, None, None, None)
```

Comment:
let's get this from the insrtument. we need to use FMT 1,1 for that !!! as opposed to FMT 1,0. So rememeber to bring the format back to it's original state after the IV sweep.

Reply:
Using FMt1,1 now. Also this class has now been moved to b1500_base